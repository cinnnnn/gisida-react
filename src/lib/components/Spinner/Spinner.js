import React from 'react';
import { connect } from 'react-redux';

require('./Spinner.scss');

const mapStateToProps = (state, ownProps) => {
  const mapId = ownProps.mapId || 'map-1';
  const MAP = state[mapId] || { blockLoad: true };
  return {
    mapId,
    MAP,
  }
}

export class Spinner extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    return (
      <div
        className="spinner"
        id={`spinner-${this.props.mapId}`}
        style={{ display: this.props.MAP.isRendered && this.props.MAP.isLoaded ? 'none' : 'block'}} />
    );
  }
}

export default connect(mapStateToProps)(Spinner);