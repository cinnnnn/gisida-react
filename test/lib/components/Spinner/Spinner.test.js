import React from 'react';
import { Spinner }  from '../../../../src/lib/components/Spinner/Spinner.js'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

//declare certain variables
const componentWrapper = shallow(
	//seems like JSX goes in here 
	<Spinner />
);

describe('Spinner', () => {
	it('Spinner component renders correctly', () => {
		const json = toJson(componentWrapper)
		expect(json).toMatchSnapshot();
	})
});