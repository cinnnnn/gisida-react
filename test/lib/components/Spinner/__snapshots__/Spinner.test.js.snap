// Jest Snapshot v1, https://goog.gl/fbAQLP

exports[`Spinner Spinner component renderes correctly 1`] = `
<div
        className="spinner"
        id={'`spinner-${this.props.mapId}`'}
        style={{ display: this.props.MAP.showSpinner ? 'block' : 'none'}} />
`;

//values are not updated